### PR TITLE
Migrate all usages of `task: UseDotNet@2` to template `install-net.yml`

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This PR contributes to:

- #5009

It is a follow-up to:
- #5010

It also remove unused `SkipDotNetInstall` parameter from the `archetype-sdk-tool-dotnet`.

### CODE REVIEW

File: eng/pipelines/apiview-review-gen-swagger.yml [hunk 1]
```diff
@@ -1,45 +1,44 @@
...
+  variables:
+  - template: /eng/pipelines/templates/variables/globals.yml
+
+  steps:
+  - script: |
+      mkdir SwaggerApiParser
+    workingDirectory: $(Pipeline.Workspace)
+    displayName: Setup working directory for Swagger Parser.
+
+  - template: /eng/pipelines/templates/steps/install-dotnet.yml
```
Comment (konrad-jamrozik) [inline eng/pipelines/apiview-review-gen-swagger.yml:25]: FYI this is the actual change

File: eng/pipelines/apiview-review-gen-swagger.yml [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
+    parameters:
+      DotNetCoreVersion: ${{ parameters.DotNetCoreVersion }}
```
Comment (konrad-jamrozik) [inline eng/pipelines/apiview-review-gen-swagger.yml:27]: FYI this is the actual change

Comment (weshaggard) [reply eng/pipelines/apiview-review-gen-swagger.yml:27]: I don't see this parameter defined in this yml where is it expected to be set?

File: src/dotnet/Mgmt.CI.BuildTools/ci.yml [hunk 3]
```diff
@@ -26,11 +26,9 @@ stages:
...
 
         steps:
-          - task: UseDotNet@2
-            displayName: 'Use .NET Core sdk $(DotNetCoreVersion)'
-            inputs:
-              packageType: sdk
-              version: $(DotNetCoreVersion)
+          - template: /eng/pipelines/templates/steps/install-dotnet.yml
+            parameters:
+              DotNetCoreVersion: $(DotNetCoreVersion)
```
Comment (weshaggard) [inline src/dotnet/Mgmt.CI.BuildTools/ci.yml:31]: We should let this use the default version and we should even consider removing DotNetCoreVersion from globals.yml

File: eng/pipelines/apiview-review-gen-swagger.yml (hunk 2 above)
Comment (weshaggard) [reply eng/pipelines/apiview-review-gen-swagger.yml:27]: I think you just want to use the default version here so remove passing this.

File: .azure-pipelines/apiview.yml [hunk 4]
```diff
@@ -165,11 +165,10 @@ stages:
...
             displayName: 'Start Azurite'
 
-          - task: UseDotNet@2
-            displayName: 'Install .NET'
-            inputs:
-              version: '$(DotNetRuntimeVersion)'
-          
+          - template: /eng/pipelines/templates/steps/install-dotnet.yml
+            parameters:
+              DotNetCoreVersion: $(DotNetRuntimeVersion)
```
Comment (weshaggard) [inline .azure-pipelines/apiview.yml:170]: Do we need to pass this here given it aligns with the default? We should consider removing the DotNetRuntimeVersion property completely. 

cc @praveenkuttappan

File: eng/pipelines/apiview-review-gen-swagger.yml (hunk 2 above)
Comment (konrad-jamrozik) [reply eng/pipelines/apiview-review-gen-swagger.yml:27]: My bad, copy-paste error. Thanks for catching this!

File: src/dotnet/Mgmt.CI.BuildTools/ci.yml (hunk 3 above)
Comment (konrad-jamrozik) [reply src/dotnet/Mgmt.CI.BuildTools/ci.yml:31]: Got it. I will investigate removal of `DotNetCoreVersion` from `globals.yml` in a separate PR.

File: .azure-pipelines/apiview.yml (hunk 4 above)
Comment (konrad-jamrozik) [reply .azure-pipelines/apiview.yml:170]: @weshaggard @praveenkuttappan the variable is [hardcoded to 6.x](https://github.com/Azure/azure-sdk-tools/blob/c1fcc8f3687828009bf5653bb225752f80fe85c4/.azure-pipelines/apiview.yml#L32). It is used in two places: here, and for [PublishTestResult@2 / TestRunTitle](https://github.com/Azure/azure-sdk-tools/blob/c1fcc8f3687828009bf5653bb225752f80fe85c4/.azure-pipelines/apiview.yml#L206):  

``` yml
- task: PublishTestResults@2
  condition: succeededOrFailed()
  inputs:
    testResultsFiles: '**/*.trx'
    testRunTitle: 'Windows DotNet $(DotNetRuntimeVersion)'
    testResultsFormat: 'VSTest'
    mergeTestResults: true
```

Removing the variable would probably necessitate removing the .NET version from the `testRunTitle`, unless you folks have some idea how to pull it into this file out from [`install-dotnet.yml`](https://github.com/Azure/azure-sdk-tools/blob/5101537df19babeffbb21eb00843f25cb0b018a4/eng/pipelines/templates/steps/install-dotnet.yml) or from [`global.json`](https://github.com/Azure/azure-sdk-tools/blob/5101537df19babeffbb21eb00843f25cb0b018a4/global.json).

But I guess it is not a big loss if we remove the version from here?

Comment (konrad-jamrozik) [reply .azure-pipelines/apiview.yml]: OK I went ahead [and removed the version variable](https://github.com/Azure/azure-sdk-tools/pull/5045/commits/baa32b74a6d1ec2211e5809bc59c5e4603d11a7d).

File: src/dotnet/Mgmt.CI.BuildTools/ci.yml (hunk 3 above)
Comment (konrad-jamrozik) [reply src/dotnet/Mgmt.CI.BuildTools/ci.yml]: @weshaggard I went ahead and [removed it in this PR](https://github.com/Azure/azure-sdk-tools/pull/5045/commits/baa32b74a6d1ec2211e5809bc59c5e4603d11a7d).

File: .azure-pipelines/apiview.yml (hunk 4 above)
Comment (weshaggard) [reply .azure-pipelines/apiview.yml]: Yes I would suggest to just remove the version from the title.

File: eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml [hunk 5]
```diff
@@ -18,9 +18,6 @@ parameters:
   - name: DotNetCoreVersion
     type: string
     default: ''
-  - name: SkipDotNetInstall
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml:21]: Where there any usages of this?

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml (hunk 5 above)
Comment (konrad-jamrozik) [reply eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml:21]: @weshaggard yes, but they were unnecessary, and so I removed them in this PR:
- #4937

**Comment by konrad-jamrozik** [comment]
Overriding, as the APIView error is unrelated:

```
[xUnit.net 00:00:21.32]     APIViewIntegrationTests.ReviewManagerTests.AddRevisionAsync_Computes_Headings_Of_Sections_With_Diff_A [FAIL]
  Failed APIViewIntegrationTests.ReviewManagerTests.AddRevisionAsync_Computes_Headings_Of_Sections_With_Diff_A [2 s]
  Error Message:
   Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: NotFound (404); Substatus: 1013; ActivityId: 6ac47a94-9bf8-43d6-848a-d7e4e9e07b32; Reason: (Collection is not yet available for read. Please retry in some time.
```

@praveenkuttappan FYI

**Pull request merged**